src/models/room.ts

```diff
@@ -2151,14 +2151,15 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         // a reference to the cached receipts anymore.
         this.cachedThreadReadReceipts.delete(threadId);
 
+        // If we managed to create a thread and figure out its `id` then we can use it
+        this.threads.set(thread.id, thread);
+
```

Comment:
So if i remember correctly the fix works because of the order of execution. 
This needs to happen before another thing, so another code path manages to retrieve the thread model?

Adding a couple of lines about that flow will probably saves us a few gray hair in the future